### ISSUE TITLE
Sphinx fails: AttributeError: 'module' object has no attribute 'getdoc'

### ISSUE THREAD

**Issue by hoechenberger** [numpy/numpydoc#26]
Hello,

this is my first time using Sphinx, so maybe I'm doing something wrong; but I checked the docs and could not figure what might be the problem.

Minimal example code:

``` python
# test.py
#
def test(a, b):
    """This is a test docstring.

    Parameters
    ----------
    a, b : int
        Some integers.

    Returns
    -------
    int
        The product of `a` and `b`.

    """

    return a*b
```

Running `sphinx-apidoc` is working fine and generating `.rst` files as expected.

However, when trying to run `make html` (or any other builder), Sphinx fails:

```
D:\Development\Testing\numpydoc-test\doc>make html
Making output directory...
Running Sphinx v1.2.2
loading pickled environment... not yet created
building [html]: targets for 3 source files that are out of date
updating environment: 3 added, 0 changed, 0 removed
reading sources... [ 33%] index
reading sources... [ 66%] modules
reading sources... [100%] test

Exception occurred:
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-p
ckages\numpydoc\numpydoc.py", line 102, in mangle_signature
    doc = SphinxDocString(pydoc.getdoc(obj))
AttributeError: 'module' object has no attribute 'getdoc'
```

My conf.py only loads the `autodoc` and `numpydoc` extensions:

```
extensions = ['sphinx.ext.autodoc', 'numpydoc']
```

This is using `Sphinx 1.2.2` and `numpydoc 0.5` on Windows 7 (64-bit) with Enthought Canopy, 32-bit.

This is the complete backtrace from the logfile:

```
# Sphinx version: 1.2.2
# Python version: 2.7.6
# Docutils version: 0.11 release
# Jinja2 version: 2.7.3
# Loaded extensions:
#   sphinx.ext.autodoc from C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\ext\autodoc.pyc
#   numpydoc from C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\numpydoc\__init__.pyc
#   sphinx.ext.oldcmarkup from C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\ext\oldcmarkup.pyc
Traceback (most recent call last):
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\cmdline.py", line 254, in main
    app.build(force_all, filenames)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\application.py", line 212, in build
    self.builder.build_update()
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\builders\__init__.py", line 214, in build_update
    'out of date' % len(to_build))
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\builders\__init__.py", line 234, in build
    purple, length):
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\builders\__init__.py", line 134, in status_iterator
    for item in iterable:
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\environment.py", line 477, in update_generator
    self.read_doc(docname, app=app)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\environment.py", line 624, in read_doc
    pub.publish()
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\core.py", line 217, in publish
    self.settings)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\readers\__init__.py", line 72, in read
    self.parse()
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\readers\__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\__init__.py", line 172, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\states.py", line 170, in run
    input_source=document['source'])
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\statemachine.py", line 239, in run
    context, state, transitions)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\states.py", line 2726, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\states.py", line 395, in new_subsection
    node=section_node, match_titles=True)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\states.py", line 282, in nested_parse
    node=node, match_titles=match_titles)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\statemachine.py", line 239, in run
    context, state, transitions)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\states.py", line 2299, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\states.py", line 2311, in explicit_construct
    return method(self, expmatch)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\states.py", line 2054, in directive
    directive_class, match, type_name, option_presets)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\docutils\parsers\rst\states.py", line 2103, in run_directive
    result = directive_instance.run()
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\ext\autodoc.py", line 1401, in run
    documenter.generate(more_content=self.content)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\ext\autodoc.py", line 768, in generate
    self.document_members(all_members)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\ext\autodoc.py", line 692, in document_members
    check_module=members_check_module and not isattr)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\ext\autodoc.py", line 755, in generate
    sig = self.format_signature()
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\ext\autodoc.py", line 939, in format_signature
    return Documenter.format_signature(self)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\ext\autodoc.py", line 423, in format_signature
    self.object, self.options, args, retann)
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\application.py", line 404, in emit_firstresult
    for result in self.emit(event, *args):
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\sphinx\application.py", line 400, in emit
    results.append(callback(self, *args))
  File "C:\Users\hoechenberger\AppData\Local\Enthought\Canopy32\User\lib\site-packages\numpydoc\numpydoc.py", line 102, in mangle_signature
    doc = SphinxDocString(pydoc.getdoc(obj))
AttributeError: 'module' object has no attribute 'getdoc'
```

Any help would be greatly appreciated!

--Richard

PS: I know this kind of posting should probably go to a mailing list, but I couldn't find any specifically for numpydoc.

**Comment by pwm1234** [comment]
I just had the same problem trying to build the sphinx documentation for AstroPy using my windows canopy python distribution. Your workaround of removing the offending pydoc.py scripts worked (but I also had to remove pydoc*.pyc files too). I submitted a bug report to Enthought. Did you?

**Comment by jdmarch** [comment]
Thanks, the fix is already in the Enthought Canopy python masters; I'm checking now to see which Canopy release includes it.

**Comment by stefanv** [comment]
This has been fixed in EPD (not sure if it has been released yet).

**Comment by jdmarch** [comment]
Not in Canopy 1.5.2 (released 30 Jan). Should be in Canopy 1.5.3 (not yet scheduled).

**Comment by jdmarch** [comment]
This has been fixed in Canopy 1.5.4. Note that in-app update is not available from Canopy 1.5.1 and earlier (uninstall + re-install is required in that case, see [this article](https://support.enthought.com/hc/en-us/articles/204469570-Canopy-shows-no-updates-available-reinstalling-from-the-website)).
